libpod/container_internal_linux.go

```diff
@@ -316,6 +319,19 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
+	// Add overlay volumes
+	for _, overlayVol := range c.config.OverlayVolumes {
+		contentDir, err := overlay.TempDir(c.config.StaticDir, c.RootUID(), c.RootGID())
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create TempDir in the %s directory", c.config.StaticDir)
+		}
+		overlayMount, err := overlay.Mount(contentDir, overlayVol.Source, overlayVol.Dest, c.RootUID(), c.RootGID(), c.runtime.store.GraphOptions())
```

Comment:
It looks like this discards the options from the overlayVol?

Reply:
the overlay pkg has hardcoded options, @rhatdan can other options be used here?

Reply:
Theoretically yes, they could be added. I guess you could specify RO, but a readonly overlay makes little sense.
The private, shared, slave might be able to be passed down.
For now we could throw an error when they are passed in, and wait for someone to complain.  I am not sure how well these options would work.

Reply:
I'd be fine with not allowing options at all here - just remove them from the struct, and error at the command line when someone tries to pass them.

Reply:
(We can always re-add them later)